WeatherService/src/main/java/org/energygrid/east/weatherservice/service/WeatherService.java

```diff
@@ -26,24 +34,69 @@ public WeatherService() {
         this.headers = new HttpHeaders();
     }
 
-    private JsonObject getCurrentWeatherData(String city){
-        //test
-        UriComponentsBuilder builder = UriComponentsBuilder.fromHttpUrl(url + "&locatie=" + city);
+    private JsonObject getWeatherData(String url){
+        UriComponentsBuilder builder = UriComponentsBuilder.fromHttpUrl(url);
 
         HttpEntity<?> entity = new HttpEntity<>(headers);
         ResponseEntity<String> response = template.exchange(builder.toUriString(), HttpMethod.GET, entity, String.class);
 
         return new Gson().fromJson(response.getBody(), JsonObject.class);
     }
 
+    public String currentWeatherStringBuilder(Point2D.Double coordinates){
+        return url + "weather?lat=" + coordinates.getX() + "&lon=" + coordinates.getY()+ "&appid=" + apiKey;
+    }
+
+    public String weatherForecastStringBuilder(Point2D.Double coordinates){
+        return url + "onecall?lat=" + coordinates.getX() + "&lon=" + coordinates.getY() + "&exclude=current,minutely,hourly,alerts" + "&appid=" + apiKey;
+    }
+
+    public double kelvinToCelsius(double kelvin){
+        return kelvin - 273.15;
+    }
+
+    public String formatDate(long unix){
+        Date date = new Date(unix*1000);
+        SimpleDateFormat format = new SimpleDateFormat("dd-MM");
+
+        return format.format(date);
+    }
+
     @Override
-    public Weather getCurrentWeather(String city) {
-        var fullWeather = getCurrentWeatherData(city);
+    public Weather getCurrentWeather(Point2D.Double coordinates) {
+        var fullWeather = getWeatherData(currentWeatherStringBuilder(coordinates)).getAsJsonObject();
+
+        var symbol = fullWeather.get("weather").getAsJsonArray().get(0).getAsJsonObject().get("icon").getAsString();
+        var tempKelvin = fullWeather.get("main").getAsJsonObject().getAsJsonObject().get("temp").getAsDouble();
+
+        var temp = Math.round(kelvinToCelsius(tempKelvin));
+
+        return new Weather(temp, symbol);
+    }
+
+    @Override
```

Comment:
22 lines method is eigenlijk te lang, is dit nodig?

Reply:
jazeker